File: app/src/main/java/org/supla/android/LocationCaptionEditor.java
package org.supla.android;

import android.content.Context;

import org.supla.android.db.DbHelper;
import org.supla.android.db.Location;
import org.supla.android.lib.SuplaClient;

public class LocationCaptionEditor extends CaptionEditor {
    public LocationCaptionEditor(Context context) {
        super(context);
    }

    @Override
    protected int getTitle() {
        return R.string.location_name;
    }

    @Override
    protected String getCaption() {
        DbHelper dbH = DbHelper.getInstance(getContext());
        Location location = dbH.getLocation(getId());
        if (location != null && location.getCaption() != null) {
            return location.getCaption();
        }
        return "";
    }

    @Override
    protected void applyChanged(String newCaption) {
        DbHelper dbH = DbHelper.getInstance(getContext());
        Location location = dbH.getLocation(getId());
        if (location != null) {
            location.setCaption(newCaption);
            dbH.updateLocation(location);
        }

        SuplaClient client = SuplaApp.getApp().getSuplaClient();
        if (client != null) {
            client.setLocationCaption(getId(), newCaption);
        }
    }
}

File: app/src/main/java/org/supla/android/LocationCaptionEditor.kt
package org.supla.android

import android.content.Context
import org.supla.android.db.DbHelper

class LocationCaptionEditor(context: Context?) : CaptionEditor(context) {

  var captionChangedListener: () -> Unit = { }

  override fun getTitle(): Int {
    return R.string.location_name
  }

  override fun getCaption(): String {
    val dbH = DbHelper.getInstance(context)
    val location = dbH.getLocation(id)
    return if (location != null && location.caption != null) {
      location.caption
    } else ""
  }

  override fun applyChanged(newCaption: String) {
    val dbH = DbHelper.getInstance(context)
    val location = dbH.getLocation(id)
    if (location != null) {
      location.caption = newCaption
      dbH.updateLocation(location)
    }
    val client = SuplaApp.getApp().getSuplaClient()
    client?.setLocationCaption(id, newCaption)
    captionChangedListener()
  }
}